docs/examples/plot_2D_basis_function.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+
+"""
+# Plotting 2D Basis Functions
+
+## Defining a 2D Basis Function
+
+To model the response of a neuron to the position of an animal in an arena, we need a 2D model for stimuli.
+In this case, the stimuli are the 2D coordinates (x, y) representing the animal's position at each time step.
+
+We can model a response function for multi-dimensional stimuli as the external product of two 1D basis functions.
+The response function approximation for the 2D case would be:
+
+$$
+f(x, y) \\approx \sum_{ij} \\alpha_{ij} \, a_i (x) b_j(y).
```

Comment:
why is this approximate?

Reply:
I am thinking about the problem as that of identifying a non-linear response function that lives in some Hilbert space of some sort. In that case, by having a finite basis set we are likely approximating the function.

for example: a basis for L2 functions is necessarily countable infinite, otherwise it will be incomplete.